src/pudl_archiver/zenodo/api_client.py

```
@@ -0,0 +1,469 @@
+        """
+        for name, resource in resources.items():
+            if name not in self.deposition_files:
+                self.changed = True
+                logger.info(f"Adding {name} to deposition.")
+
+                with open(resource.local_path, "rb") as f:
+                    await self.upload(f, resource.local_path.name)
+            else:
+                file_info = self.deposition_files[name]
+
+                # If file is not exact match for existing file, update with new file
+                if not _compute_md5(resource.local_path) == file_info.checksum:
+                    self.changed = True
+                    logger.info(f"Updating {name}")
+                    await self.delete_file(file_info)
+
+                    with open(resource.local_path, "rb") as f:
+                        await self.upload(f, resource.local_path.name)
+
+        # Delete files not included in new deposition
+        for filename, file in self.deposition_files.items():
+            if filename not in resources and filename != "datapackage.json":
+                self.changed = True
+                await self.delete_file(file)
+
+        await self.update_datapackage(resources)
+
+    async def get_deposition(self, concept_doi: str):
+        """Get data for a single Zenodo Deposition based on the provided query.
+
+        See https://developers.zenodo.org for more information.
+
+        Args:
+            concept_doi: Concept DOI for desired data source.
+
+        Returns:
+            Deposition metadata pertaining to the latest version returned by Zenodo api.
+        """
+        url = f"{self.api_root}/deposit/depositions"
+        params = {"q": f'conceptdoi:"{concept_doi}"', "access_token": self.upload_key}
+
+        async with self.session.get(url, params=params) as response:
+            # Deposition should be first element in json response
+            raw_json = await response.json()
+
+            return Deposition(**raw_json[0])
```

Can we raise an exception here if the first element of `raw_json` doesn't match our expectations?

---

Added a check for the array length, and I think any other errors will get caught in the pydantic validation of the `Deposition` object

---

Oh that's right. Thanks pydantic.